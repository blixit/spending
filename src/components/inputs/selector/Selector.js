import React from 'react';
import styled from 'styled-components';

const StyledSelect = styled.select`
  min-height: 25px;
  lineheight: 25px;
  font-size: 16px;
`;

function Selector (props) {
  const { values, onChange } = props;
  return (
    <StyledSelect onChange={onChange} {...props}>
      {values.map(
        value => (<option key={String(value)}>{value}</option>)
      )}
    </StyledSelect>
  );
}

export default Selector;
