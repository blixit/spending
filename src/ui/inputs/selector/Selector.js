import React from 'react';
import styled from 'styled-components';

const StyledSelect = styled.select`
  min-height: 30px;
  lineheight: 25px;
  font-size: 0.8rem;
  border: 1px solid silver;
  border-radius: 3px;
  background-color: transparent;

  &:focus{
    outline: none;
    border: 1px solid ${({ theme, color }) => color ? theme[color]() : '' }
    background-color: ${({ theme, color }) => color ? theme[color]() : '' }
    color: ${({ theme, color }) => color ? theme.text[color]() : '' }  
  }
`;

function Selector (props) {
  const { values, selected, ...rest } = props;
  return (
    <StyledSelect value={selected} {...rest}>
      {values.map(
        ({ id, option }) => (<option key={id}>{option}</option>)
      )}
    </StyledSelect>
  );
}

export default Selector;
