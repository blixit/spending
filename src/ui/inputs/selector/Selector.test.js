import React from 'react';

import render from 'testing/render';

import Selector from './Selector';

describe('[inputs/selector] Selector', () => {
  const options = [{
    id: 1,
    option: 'item 1'
  }];
  it('renders properly', () => {
    const tree = render(
      <Selector
        label='Label'
        onChange={jest.fn()}
        values={options}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
