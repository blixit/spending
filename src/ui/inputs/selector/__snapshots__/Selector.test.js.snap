// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[inputs/selector] Selector renders properly 1`] = `
<select
  className="sc-bdVaJa kNYkjY"
  label="Label"
  onChange={[MockFunction]}
>
  <option>
    item 1
  </option>
</select>
`;
